File: Examples/intel_examples/examples_core_f/transf/source/common_func.f
!===============================================================================
! Copyright 2010-2020 Intel Corporation.
!
! This software and the related documents are Intel copyrighted  materials,  and
! your use of  them is  governed by the  express license  under which  they were
! provided to you (License).  Unless the License provides otherwise, you may not
! use, modify, copy, publish, distribute,  disclose or transmit this software or
! the related documents without Intel's prior written permission.
!
! This software and the related documents  are provided as  is,  with no express
! or implied  warranties,  other  than those  that are  expressly stated  in the
! License.
!===============================================================================

!  Content:
!
!*******************************************************************************
!   Is'nt work yet :)
      subroutine printArray(a, n, m, data_type)

      integer       m, n
      character*1   data_type
      real          a(n,*)
      integer       i, j

      if (lge(data_type,'s')) then
        do i=1,n
           print 100, (a(i,j),j=1,m)
        end do
      else if (lge(data_type,'d')) then
             do i=1,n
                print 100, (a(i,j),j=1,m)
             end do
           else if (lge(data_type,'c')) then
                    do i=1,n
                       print 110, (a(i,j),j=1,m)              
                    end do
                else if(lge(data_type,'z')) then
                      do i=1,n
                        print 110, (a(i,j),j=1,m)
                      end do
                    end if

 100  format(9x, 10(f6.2,',',f6.2,3x)) ! for d, s
 110  format(9x, 10(f8.3,2x))          ! for c, z

      return
      end

File: Examples/intel_examples/examples_core_c/transc/source/common_func.c
/*******************************************************************************
* Copyright 2010-2020 Intel Corporation.
*
* This software and the related documents are Intel copyrighted  materials,  and
* your use of  them is  governed by the  express license  under which  they were
* provided to you (License).  Unless the License provides otherwise, you may not
* use, modify, copy, publish, distribute,  disclose or transmit this software or
* the related documents without Intel's prior written permission.
*
* This software and the related documents  are provided as  is,  with no express
* or implied  warranties,  other  than those  that are  expressly stated  in the
* License.
*******************************************************************************/

/*
!  Content:
!
!******************************************************************************/
#include <mkl_types.h>
#include "common_func.h"
/* 
**  Print matrix of any type
*/                          

size_t print_number(void* data, char data_type)
{ 
  switch(data_type)
  {
    case 's':
      printf("%4f\t", *((float*)data));  
      return sizeof(float);
    case 'd':
      printf("%4lf\t", *((double*)data));  
      return sizeof(double);
    case 'c':
      printf("(%4f,%4fi)\t", ((MKL_Complex8*)data)->real, ((MKL_Complex8*)data)->imag);  
      return sizeof(MKL_Complex8);
    case 'z':
      printf("(%4lf,%4lfi)\t", ((MKL_Complex16*)data)->real, ((MKL_Complex16*)data)->imag);  
      return sizeof(MKL_Complex16);
  }
  return 0;
}

void print_matrix(size_t rows, size_t cols, char data_type, void *src)
{
  size_t i=0;
  size_t j=0;
  char* p_char=((char*) src);

  for(i=0; i<rows; i++)
  {
    for(j=0; j<cols; j++)
      p_char += print_number( p_char , data_type);
    printf("\n");
  }
}
